junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -121,12 +126,28 @@ private void assertValidFieldCandidate(Field field) {
 	 */
 	@Override
 	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
-		boolean annotated = parameterContext.isAnnotated(TempDir.class);
-		if (annotated && parameterContext.getDeclaringExecutable() instanceof Constructor) {
-			throw new ParameterResolutionException(
-				"@TempDir is not supported on constructor parameters. Please use field injection instead.");
+		TempDir annotation = parameterContext.findAnnotation(TempDir.class).orElse(null);
+		if (annotation != null) {
+			if (parameterContext.getDeclaringExecutable() instanceof Constructor) {
+				throw new ParameterResolutionException(
+					"@TempDir is not supported on constructor parameters. Please use field injection instead.");
+			}
+			boolean hasAnotherParameterWithTempDirAnnotationWithSameIdentifier = Stream.of(
+				parameterContext.getDeclaringExecutable().getParameters()) //
+					.skip(parameterContext.getIndex() + 1) //
+					.map(parameter -> findAnnotation(parameter, TempDir.class).orElse(null)) //
+					.filter(Objects::nonNull) //
+					.map(TempDir::value) //
+					.anyMatch(isEqual(annotation.value()));
```

Comment:
```suggestion
					.anyMatch(annotation::equals));
```

Since `@TempDir` only declares a single attribute, you should be able to rely on `.equals()` for the annotation instances.

Reply:
TIL about `Annotation.equals`